packages/core/src/InstanceSettings.ts

```diff
@@ -57,26 +59,44 @@ export class InstanceSettings {
 		return process.env.N8N_USER_FOLDER ?? process.env[homeVarName] ?? process.cwd();
 	}
 
+	/**
+	 * Load instance settings from the settings file. If missing, create a new
+	 * settings file with an auto-generated encryption key.
+	 */
 	private loadOrCreate(): Settings {
-		let settings: Settings;
-		const { settingsFile } = this;
-		if (existsSync(settingsFile)) {
-			const content = readFileSync(settingsFile, 'utf8');
-			settings = jsonParse(content, {
-				errorMessage: `Error parsing n8n-config file "${settingsFile}". It does not seem to be valid JSON.`,
+		if (existsSync(this.settingsFile)) {
+			const content = readFileSync(this.settingsFile, 'utf8');
+
+			const settings = jsonParse<Settings>(content, {
+				errorMessage: `Error parsing n8n-config file "${this.settingsFile}". It does not seem to be valid JSON.`,
 			});
-		} else {
-			// Ensure that the `.n8n` folder exists
-			mkdirSync(this.n8nFolder, { recursive: true });
-			// If file doesn't exist, create new settings
-			const encryptionKey = process.env.N8N_ENCRYPTION_KEY ?? randomBytes(24).toString('base64');
-			settings = { encryptionKey };
-			this.save(settings);
-			// console.info(`UserSettings were generated and saved to: ${settingsFile}`);
+
+			if (!inTest) console.info(`User settings loaded from: ${this.settingsFile}`);
+
+			const { encryptionKey, tunnelSubdomain } = settings;
+
+			if (process.env.N8N_ENCRYPTION_KEY && encryptionKey !== process.env.N8N_ENCRYPTION_KEY) {
+				throw new ApplicationError(
```

Comment:
Existing behavior is to take precedence in the config file - should we throw and prevent starting or just warn the users about it?

Reply:
I was following [this](https://n8nio.slack.com/archives/C035KBDA917/p1704205654922949?thread_ts=1704205228.950219&cid=C035KBDA917).